### ISSUE TITLE
Using ssh controlmaster breaks terminal when using libvirt and port forwards

### ISSUE THREAD

**Issue by jhalfmoon** [hashicorp/vagrant#10372]
### Vagrant version
2.2.0

### Host operating system
Arch Linux

### Guest operating system
Centos 7.5 (Bento)

### Vagrantfile
```ruby
Vagrant.configure("2") do |config|
    config.vm.box = "bento/centos-7.5"
    config.vm.define "node1", autostart: true, primary: true do |node|
      node.vm.network "forwarded_port", guest: 8080, host: 18080
    end
end
```

### Debug output
https://gist.github.com/jhalfmoon/00179bf8ee21b3a2570efd8522f28d1d

### Expected behavior
After running Vagrant and returning to the command prompt, the terminal should work as usual.

### Actual behavior
Each second keystroke 'disappear', making it near impossible to use the terminal. After destroying the vm, the behavior of the terminal will return to normal.

### Steps to reproduce
- Setup your ssh client to use controlmaste:
```
# ~/.ssh/config
Host *
ControlMaster yes
```
- Install a working libvirt/kvm environement and activate it:
```
export VAGRANT_DEFAULT_PROVIDER=libvirt
```
- Run vagrant with the given Vagrantfile
- The terminal used to start vagrant will now have the issue
- Doing 'vagrant destroy' will restore normal behaviour
- Repeating the above experiment with 'ControlMaster no' will not exhibit any issues

Application versions uses:
- OpenSSH_7.9p1, OpenSSL 1.1.1  11 Sep 2018
- GNU bash, version 4.4.23(1)

### References
None, as far as I can tell.

**Comment by chrisroberts** [comment]
@jhalfmoon If you disable the default synced folder so nothing is synced to the new guest, do you still experience the same behavior in your shell?

```ruby
config.vm.synced_folder ".", "/vagrant", disabled: true
```

**Comment by jhalfmoon** [comment]
Yes, the same behaviour remains. I also tried to see if the issue is maybe related to my terminal and/or tmux, which I always use. So I used xfce-terminal instead of Konsole and also ran it without tmux, but the issue remains the same. It's as if something is stealing characters from stdin. An strace of the bash process doesn't shine any light on this either as there's no difference to be seen between the working and non-working situation.
